sdk/go/common/workspace/project.go

```diff
@@ -97,6 +98,18 @@ type Plugins struct {
 	Analyzers []PluginOptions `json:"analyzers,omitempty" yaml:"analyzers,omitempty"`
 }
 
+type ProjectConfigItemsType struct {
+	Type  string                  `json:"type" yaml:"type"`
+	Items *ProjectConfigItemsType `json:"items" yaml:"items"`
+}
```

Comment:
Would it be possible to express this as a `schema.Type`? We already have a lot of logic to work on these, and we will need to translate them into schema types anyways for entry into the Pulumi YAML type checker.

Reply:
It's possible to write a function to translate a config type into `schema.Type` but I wouldn't want to start with this right off the bat and only support what is needed because it would seem that a config type can be any `schema.Type` but that is not the case

Reply:
Agree with Zaid here, config types != schema types, it will _probably_ end up being a subset but possibly a non-intersecting set.

Reply:
We will definitely need the `config type -> schema type` function when we incorporate this change into Pulumi YAML. We can wait for this change to settle before we unify.

Reply:
That's a good point that this will probably have to be a subset otherwise there would be config values that YAML wouldn't be able to use, but I think given the subset nature of it I'm going to continue err'ing towards us not making it a schema.Type for now but it's own thing.

Reply:
Actually see above, I think it might end up being non-intersecting still if we end up with config types that YAML doesn't support.